### ISSUE TITLE
Closing unpaid AR/AP > Transaction or Invoice

### ISSUE THREAD

**Issue by aungzwin10** [ledgersmb/LedgerSMB#6007]
As we discussed in the channel about my use-case of closing the AR/AP transaction or invoice which doesn't need to make payment anymore, I have tried as @ehuelsmann said 
```
-- you can create a VOID invoice by clicking Void on the invoice. Then you have an invoice with the same items, but negative. Then post that. The result is 2 open invoices, the original and the void. you can make a zero payment to close the two invoices.

-- Oh! Transaction ; that doesn't have void. I thought invoice (which does). In that case: create a transaction with all negative values, post that and then do the zero-payment step.
```

but it doesn't make any changes. Please see below screenshot for your reference.

https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/Pg-database.sql#L1816-L1818

As I researched in the sql, I found that there is a column named `force_closed` in the `ar, ap and eca_invoice` table.
The comment on the column seem to be same with my requirement. But I didn't see any place which use this column in the code.
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/Pg-database.sql#L1816-L1818

**Comment by ehuelsmann** [comment]
The `force_closed` column is definitely in use in our code-base, as shown below. However, the use of this field has at least one very bad side-effect: the invoice remains on the AP account, but won't appear in the outstanding transactions list, which means that the outstanding report no longer reconciles with the AP acount total, but if the amount is actually not owed anymore, then it should also be cleared from the balance. To prevent these negative side effects, I proposed the solution of the negative transaction. Can you show what you mean by "it doesn't make changes"?

https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L124
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L136
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L164
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L446
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L453
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L488
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L558
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L566
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L626
https://github.com/ledgersmb/LedgerSMB/blob/2d8cc1c09bae40bc6c39bfc2c14782ae7792d517/sql/modules/Report.sql#L630

**Comment by aungzwin10** [comment]
<img width="1208" alt="Screen Shot 2021-09-20 at 11 45 18 AM" src="https://user-images.githubusercontent.com/31498748/134104757-fa665bff-efbf-43e0-8fa4-88874345a0db.png">

**Comment by aungzwin10** [comment]
Sorry, I forgot to add the screenshot. As you see in the below screenshot, I created negative transaction.
Cash > Payment show these two invoices, I changed `To pay` amount to `0` then Post. But both invoices remain as outstanding and it still appear in the Cash > Payment. Invoice will disappear from the Cash > Payment once the invoice is closed, and that invoice will appear when filtering with `Closed` invoices in AR/AP > Search, no?

**Comment by ehuelsmann** [comment]
@aungzwin10, thanks for posting the images: as before, I now understand where our conversation went wrong.

What I meant is that one line is entered at 432 and the other at -432. Then the Subtotal and Total lines will show "0.00" (which is what I was referring to as "a zero-payment": the total amount of the payment is exactly zero). When you do that, both invoices get closed.

**Comment by aungzwin10** [comment]
@ehuelsmann you mean that, I need to do the payment as it original outstanding amount for both invoices. So the payment lines will be recorded for both invoice in the journal? 
And the amount 432 will not sum to profit nor expense, right?

**Comment by ehuelsmann** [comment]
@aungzwin10 yes: if you do all amounts on the same accounts with opposite signs, then yes, that's what will be the effect.

**Comment by ehuelsmann** [comment]
This is - now - a duplicate of  #5863 (discussions).
